src/Couchbase.Lite.Shared/API/Document/Document.cs

```diff
@@ -96,6 +96,12 @@ internal C4DocumentWrapper c4Doc
         [CanBeNull]
         internal Database Database { get; set; }
 
+        /// <summary>
+        /// Gets the Collection that this document belongs to, if any
+        /// </summary>
+        [CanBeNull]
+        public Collection Collection { get; set; }
```

Comment:
ICollection (I didn't make an exhaustive list, but if we have these interfaces anyway let's use them)